src/main/java/org/openimmunizationsoftware/cdsi/core/logic/CreateRelevantPatientSeries.java

```diff
@@ -0,0 +1,115 @@
+package org.openimmunizationsoftware.cdsi.core.logic;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openimmunizationsoftware.cdsi.core.data.DataModel;
+import org.openimmunizationsoftware.cdsi.core.domain.Antigen;
+import org.openimmunizationsoftware.cdsi.core.domain.AntigenSeries;
+import org.openimmunizationsoftware.cdsi.core.domain.PatientSeries;
+
+public class CreateRelevantPatientSeries extends LogicStep {
+
+  public CreateRelevantPatientSeries(DataModel dataModel) {
+    super(LogicStepType.CREATE_RELEVANT_PATIENT_SERIES, dataModel);
+  }
+
+  @Override
+  public LogicStep process() throws Exception {
+
+    if (dataModel.getAntigenSelectedList() == null) {
```

Comment:
I generally like the rule of no triple nests, I find it tends to make the code much less readable you might want to consider extracting some of the nesting components into thier own helper methods